### PR TITLE
Fix app event behaviour and iOS build fix

### PR DESCRIPTION
**Pull request by themuflon**

### CODE REVIEW

File: loom/engine/bindings/loom/lmApplicationTasks.cpp [hunk 1]
```diff
@@ -36,11 +36,26 @@ extern "C"
...
 
 void loom_tick()
 {
-    if (atomic_load32(&gLoomTicking) < 1) return;
-    
+    if (atomic_load32(&gLoomTicking) < 1)
+    {
+
+        // Signal that the app has really stopped execution
+        lmLogWarn(gTickLogGroup, "Paused: %d", atomic_load32(&gLoomPaused));
```
Comment (SmilyOrg) [inline loom/engine/bindings/loom/lmApplicationTasks.cpp]: Won't this print every tick while the app is paused? Assuming it gets scheduled time.

Comment (themuflon) [inline loom/engine/bindings/loom/lmApplicationTasks.cpp]: Don't bother yet, debug stuff :)

Comment (SmilyOrg) [inline loom/engine/bindings/loom/lmApplicationTasks.cpp]: Ah :)

File: application/common/core.cpp [hunk 2]
```diff
@@ -71,11 +71,12 @@ void loop()
     /* Check for events */
     while (SDL_PollEvent(&event))
     {
-        if (event.type == SDL_QUIT)
+        if (event.type == SDL_QUIT ||
+            event.type == SDL_APP_TERMINATING)
```
Comment (SmilyOrg) [inline application/common/core.cpp]: Maybe this would be better suited for the priority filter around line 300: `sdlPriorityEvents`

Comment (themuflon) [inline application/common/core.cpp]: I thought about that too, but it wouldn't really help much. Even if it pops up in another thread (on some devices), you still have to wait for the tick to finish or force an exit.

Comment (SmilyOrg) [inline application/common/core.cpp]: Ah right, because we process all SDL events before the next tick anyway.

Comment (SmilyOrg) [inline application/common/core.cpp]: Might be good to have a short comment on that above this line?

File: loom/engine/bindings/loom/lmApplication.cpp [hunk 3]
```diff
@@ -87,6 +88,16 @@ void loom_appPause(void)
 {
     int ticking = atomic_decrement(&gLoomTicking);
     if (ticking != 0) return;
+
+    // Wait for the main thread to stop all GL execution
+    // if where on a different thread
```
Comment (SmilyOrg) [inline loom/engine/bindings/loom/lmApplication.cpp]: typo: "if where" -> "if we're"

**Comment by SmilyOrg** [comment]
Tiny nit, then we can land!

**Comment by SmilyOrg** [comment]
This seems good now!

**Pull request merged**